### ISSUE TITLE
quirk FOCUSPREV doesn't update border color

### ISSUE THREAD

**Issue by wavexx** [conformal/spectrwm#87]
When setting the FOCUSPREV quirk on a window, and the window is unmapped, the focus is given to the previous window but the border color isn't updated accordingly.

**Comment by LordReg** [comment]
Hi wavexx, I tested various programs and could not reproduce the issue.  Would you give an example of a program and corresponding quirk entry where you experience the problem?

**Comment by wavexx** [comment]
On 01/13/2015 06:30 AM, LordReg wrote:
> Hi wavexx, I tested various programs and could not reproduce the
> issue.  Would you give an example of a program and corresponding
> quirk entry where you experience the problem?

For a reproducible test case:

https://finestructure.net/assword/

with "xdo" installed and with quirk:

quirk[Assword] = FLOAT + FOCUSPREV

open a terminal, then run:

  assword gui

enter something, click "create" (or click an existing entry), and watch
the focus disappear.

It might be required to init the assword DB first.

**Comment by wavexx** [comment]
Hum, might not be strictly related to FOCUSPREV though. Looks like several java applications are also quite good at reproducing the issue without any quirk.

For example "jabref", menu -> help -> about dialog will always appear as unfocused (the main window loses focus as well as long as you have only jabref in the workspace). Yet, if you cycle the focused region, focus is assigned to the about dialog, which is the same behavior I was describing before.

**Comment by marcopeereboom** [comment]
So this sounds like java is busted again.  It periodically happens.  Someone needs to have another look and work around that pile *again*.

**Comment by wavexx** [comment]
On 02/17/2015 07:14 PM, marcopeereboom wrote:
> So this sounds like java is busted again.  It periodically happens.
> Someone needs to have another look and work around that pile
> *again*.

I don't think so.

Thunderbird still shows odd behavior from time to time when composing
new messages, and I have the same issue with some dialog windows from
time to time.

The problem I'm always busy when this happen to debug it.

**Comment by marcopeereboom** [comment]
And that is on the latest version?  We had some issues with TB a while back but i thought they were all fixed.  If you are on the latest is there some pattern you notice that gets you into that state?

**Comment by wavexx** [comment]
On 02/19/2015 03:07 PM, marcopeereboom wrote:
> And that is on the latest version?  We had some issues with TB a

Obviously on current master.

> while back but i thought they were all fixed.  If you are on the

The previous case with TB was due to the address completion popup.
This was fixed.

> latest is there some pattern you notice that gets you into that
> state?

I sometimes press Ctrl+N to compose a new message.

The new composer window appears, but cannot be focused. If I click on
it, the focus goes immediately to the next region. If there's no other
region, it goes back to the thunderbird message list.

But I discovered I can search for it using search_win+number, and in
this case the window enters the focus cycle and works regularly afterwards.

This is the same state those java dialog windows appears to be in, but
it's clearly much harder to reproduce.
